checkpoint/runner_test.go

```diff
@@ -231,89 +252,68 @@ func asAtxSnapshot(v *types.ActivationTx, cmt *types.ATXID) types.AtxSnapshot {
 	}
 }
 
-func createMesh(t *testing.T, db *sql.Database, miners map[types.NodeID][]*types.ActivationTx, accts []*types.Account) {
-	for _, vatxs := range miners {
-		vatxs = slices.Clone(vatxs)
-		// ATXs are expected to be in reverse epoch order and we want older ATXs
-		// created first so that the nonce can be retrieved from them for
-		// populating the nonce field when creating newer ones
-		slices.Reverse(vatxs)
-		for _, atx := range vatxs {
+func createMesh(t *testing.T, db *sql.Database, miners []miner, accts []*types.Account) {
+	for _, miner := range miners {
+		for _, atx := range miner.atxs {
 			require.NoError(t, atxs.Add(db, atx))
 		}
+		if proof := miner.malfeasanceProof; len(proof) > 0 {
+			require.NoError(t, identities.SetMalicious(db, miner.atxs[0].SmesherID, proof, time.Now()))
+		}
 	}
 
 	for _, it := range accts {
 		require.NoError(t, accounts.Update(db, it))
 	}
-
-	// smesher 5 is malicious and equivocated in epoch 7
-	bad := types.BytesToNodeID([]byte("smesher5"))
-	require.NoError(t, atxs.Add(db, newAtx(types.ATXID{83}, types.EmptyATXID, &types.ATXID{27}, 7, 0, 113, bad)))
-	require.NoError(t, atxs.Add(db, newAtx(types.ATXID{97}, types.EmptyATXID, &types.ATXID{16}, 7, 0, 113, bad)))
-	require.NoError(t, identities.SetMalicious(db, bad, []byte("bad"), time.Now()))
 }
 
 func TestRunner_Generate(t *testing.T) {
+	t.Parallel()
 	tcs := []struct {
 		desc    string
-		atxes   map[types.NodeID][]*types.ActivationTx
+		miners  []miner
 		numAtxs int
 		accts   []*types.Account
-		fail    bool
 	}{
 		{
 			desc:    "all good, 2 atxs",
-			atxes:   allAtxs,
+			miners:  allMiners,
 			numAtxs: 2,
 			accts:   allAccounts,
 		},
 		{
 			desc:    "all good, 4 atxs",
-			atxes:   allAtxs,
+			miners:  allMiners,
 			numAtxs: 4,
 			accts:   allAccounts,
 		},
 		{
 			desc:    "all good, 7 atxs",
-			atxes:   allAtxs,
+			miners:  allMiners,
 			numAtxs: 7,
 			accts:   allAccounts,
 		},
-		{
-			desc:  "no atxs",
-			accts: allAccounts,
-			fail:  true,
-		},
-		{
-			desc:  "no accounts",
-			atxes: allAtxs,
-			fail:  true,
-		},
```

Comment:
I'm not sure if we should remove test cases with invalid data checking if they fail as expected?

Reply:
They are not removed, but moved below